pallets/parachain-staking/src/lib.rs

```diff
@@ -1674,4 +1674,29 @@ pub mod pallet {
 			Self::selected_candidates()
 		}
 	}
+
+	pub struct InitGenesisOnRuntimeUpgrade<T>(PhantomData<T>);
```

Comment:
Run on runtime upgrade by including it here- https://github.com/OAK-Foundation/OAK-blockchain/blob/5e04b5e370633219bf3598c780c22f9559b79ab2/runtime/neumann/src/lib.rs#L113

I'm still shaky on the running of migrations and very open to other approaches here.

Reply:
yah i think this is fine?

Reply:
why is and no the hook? https://docs.substrate.io/rustdocs/latest/frame_support/traits/trait.Hooks.html#provided-methods

Reply:
No good reason. I just didn't know the pattern with the hooks. I found the pattern I'm using here- https://github.com/paritytech/substrate/tree/master/frame/executive#custom-onruntimeupgrade-logic

Reply:
Going to test out the hook, will report back.